### DISCUSSION TITLE
Ampache Develop: Random Play dissappeard

### DISCUSSION THREAD

**Discussion by SePHTaN** [ampache/ampache#2823] [Q&A]
Just realised that in develop there is no more the option to quickly start a random play for songs, albums, artists and playlists.
Is this an issue or intended to get removed.  ?

I ask because i am currently working on #1365 and wanted to test wether what i have currently done is fully working.
Album and Artist download with Art included works, so i wanted to test wether this applies to playlist download also.

So i wanted to add a playlist the quick way ... bump ... feature at least currently not available.

Second i was looking if there is a way for me to assign this open todo in Ampache Improvement and Modernization to me.
 
 When adding in things like this and regarding Contributing.md: Lines that are not part of your change should not be edited
 Is it ok to do modernization changes like:
Use Ampache\Config\ConfigContainerInterface;  
gets
use Ampache\Config\AmpConfig;

$options = ['comment' => $this->configContainer->get('file_zip_comment'), ]; 
gets
$options = ['comment' => AmpConfig::get('file_zip_comment'), ];

Another question : 
when adding a additional configuration parameter where do i have to do changes besides ampache.cfg.php
i want to add a parameter for art inclusion in zip files : art_zip_add = true/false

**Comment by lachlan-00** [comment]
random is inside the playlist rightbar now only, you have to use it to play and the sidebar is getting too big for duplicate items.
![image](https://user-images.githubusercontent.com/1305249/111087048-30b38b80-856b-11eb-8fe6-3b578f6ac1a0.png)

the advanced 'random' search is inside the search part now.
![image](https://user-images.githubusercontent.com/1305249/111087055-3f01a780-856b-11eb-95db-4edf7770375d.png)

I actually think that if you download an object like a full album it should include the art anyway so i'm not sure if you need a config item for it. But if you update the default config file you usually just bump the value up one to force a rewrite

`src/Config/Init/InitializationHandlerConfig.php`
![image](https://user-images.githubusercontent.com/1305249/111087629-660da880-856e-11eb-8f6e-3f13eafcd77a.png)

**Reply by SePHTaN** [reply]
In principle i am finished with this little enhancement.
I manged to get covers included, tested for album, artist and playlist download.
Also i add a simple_m3u file in the zip.
i've looked into what ampache does for playlist management but for my intention it was way to complicated for just adding an m3u playlist file in the zip.
So besides adding additional parameter(s) in the config i have to increase the config-file version in ampache.cfg.php.dist
and in src/Config/Init/InitializationHandlerConfig.php nowhere else.

>I actually think that if you download an object like a full album it should include the art anyway

The same wagnered mentioned this afternoon, up til now there hasent been art included in the zip, and from Ziphandler.php in can say there is nothing dealing with art inside.
I also do only include art in the zip when 'album_art_preferred_filename' exists in the referring folder, i don't pull it from database. Would be also doable.
If there is no file mostly that is because art is included in audiofiles hence not necessary in zip.

Tomorrow i will update my fork of ampache and after update my local clone i will look into merging upstream.
First time for me :)

**Reply by lachlan-00** [reply]
That's great to hear. It sounds like you're going to make a great first contribution then!
